# Assignment 2 
## Chloe Firth
## B00707576


# Using rvest and dpylr to Extract NBA Data and Find Maximum Points Per Game Values Depending on Month


## Introduction


Looking at statistics and manipulating data is something that many people do, even when they do not realize they are doing it. Being able to take data from a website and use R to extract particular data from a dataset can be used to many peoples advantages to finding the data they’re looking for. Rvest is a package that was created to take data from web pages and be able to incorporate that data from that website using the html_node() function to select the specific data of interest (Wickham, 2014). This package joined with another package called dplyr which can manipulate the data not only from HTML but any data set (Wickham, 2017) allowed this study to answer the question, which NBA players have the most points per game (PPG) during the regular season depending on the month.

A study by Bhandari et al. found that similar methods have also been done in a PC application called Advanced Scout which allows NBA couches and trainers to see patterns in different games to better improve strategy for upcoming games however the method discussed if specific for NBA data (Bhandari et al., 1997). While this approach is widely accepted throughout the NBA the idea of this study applying datasets from HTML to find specific data can be used to any data within any different sports leagues or any information of interest where this study can be applied. 

A package in R called ballr created by Ryan Elmore and Peter DeWitt was found which is could find similar information of interest however using data current and historical basketball games and allows the user to find out specific information for a particular season, player or position within their records and allows you to look at the statistics per game, year,  team,  date or in a specific time frame, per any amount of positions, or throughout a players entire career (Elmore & Dewitt, 2018). This package conducts similar research and findings within R much like this study; however, Ballr allows the user to find this information with no need to use HTML coding to import data.

The results of this study can be beneficial to not only coaches and managers of the NBA but to the fans as well. Coaches and managers can use this data when they’re trading their players to find pieces to their team that they’re missing to improve their team better or this information can be beneficial when creating plays or altering the starting lineup depending on the month to cater to the statistics found through this study. Besides this, the coaches and trainers of any NBA team will be able to use this information to encourage any different training aspects depending on the statistic found within each month. This study will also contribute toward the information that will be important to the spectators, with this information they will be able to alter their fantasy NBA draft or any strategic bets they may wish to do. Knowing this information, any individual that can make the appropriate changes to any best or selections they may want to change.


## Methods/Results

The first step is to find an appropriate website in which it can give the proper information to use and manipulate to find the information for this study. This [website](https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/points/All/desc/1/Regular_Season) was chosen because unlike other websites; it allows you to look at specific data and times such as different months or to look at the play-off or regular season,  different teams or positions can be looked at individually as well. 

The next step to this project is to acquire the necessary packages to allow the study to work appropriately. The first package is rvest which enable R to read HTML and take information from the website. The second package is dplyr which allows R to manipulate the data to join the tables together, so they’re easier to look at and easier to extract information from.

In [1]:
library(rvest)
library(dplyr)

Loading required package: xml2

Attaching package: ‘dplyr’

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



The rest of the methods will be shown through the example of the month of October:

### October

The website was then read in R using the `read_html()` function. Each month had to read four different website pages because the website only shows 100 players on each page, seeing as there are approximately 336 players in the NBA there were four pages to extract tables from. 

In [2]:
NBA1 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/October")

In [3]:
NBA2 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/October")

In [4]:
NBA3 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/October")

In [5]:
NBA4 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/October")

The `html_nodes('table')%>% .[[1]] %>% html_table()` code was used to take the first `%>% .[[1]]` table `[ html_nodes(‘table’)]` on the website out of the HTML coding and place it into R where it can be manipulated and taken information from. The last part of the code `[%>% html_table()]` will allow the table to show below. This step was completed four times, once for each page for that month.

In [6]:
October1 = NBA1 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [7]:
October2 = NBA2 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [8]:
October3 = NBA3 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [9]:
October4 = NBA4 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

Next the tables were strung together by the function `full_join()`. This function can only join two tables together at a time, this was done three times, once to join the first 200 players, then the last 136 players, and lastly those two joined tables were joined together which looks like this:

In [10]:
a = full_join(October1, October2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [11]:
b = full_join(October3, October4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [12]:
October = full_join(a, b)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 1: The data and statistics for all players in the NBA during the month of October in the 2017-2018 season

In [13]:
October

#   Player                Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  7  19.3  1.6  4.4 0.355 0.9 2.6 ⋯ 1.000 0.7
2    2  Quincy Acy            BRK  6  14.9  2.0  3.8 0.522 1.8 3.5 ⋯ 0.000 1.5
3    3  Steven Adams          OKC  7  30.7  6.4  9.4 0.682 0.0 0.0 ⋯ 0.733 1.4
4    4  Bam Adebayo           MIA  4  20.2  2.2  5.0 0.450 0.0 0.0 ⋯ 0.625 0.5
5    5  LaMarcus Aldridge     SAS  7  34.0  9.1 18.4 0.496 0.6 1.7 ⋯ 0.786 1.6
6    6  Jarrett Allen         BRK  6  15.0  1.5  3.7 0.409 0.0 0.2 ⋯ 0.600 0.5
7    7  Tony Allen            NOP  7  10.7  1.7  3.6 0.480 0.1 0.7 ⋯ 0.250 1.0
8    8  Al-Farouq Aminu       POR  7  30.0  3.4  7.7 0.444 1.6 3.7 ⋯ 1.000 0.9
9    9  Justin Anderson       PHI  5  12.0  1.6  5.0 0.320 1.0 3.0 ⋯ 0.500 0.4
10  10  Kyle Anderson         SAS  7  26.6  3.1  7.4 0.423 0.3 0.7 ⋯ 0.643 1.0
11  11  Ryan Anderson         HOU  8  31.9  4.1 10.0 0.412 3.2 8.4 ⋯ 0.778 0.5
12  12  Giannis Antetokounmpo MIL  7  37.4 13.0 20.6 0.632 0.9 2.1 ⋯ 0.774 3.3
13  13  Carmelo Anthony       OKC  7  33.4  8.4 18.1 0.465 2.4 7.0 ⋯ 0.806 1.7
14  14  Ogugua Anunoby        TOR  6  17.6  2.2  6.0 0.361 0.8 3.2 ⋯ 0.800 0.2
15  15  Trevor Ariza          HOU  6  35.6  1.8  7.5 0.244 1.0 5.3 ⋯ 1.000 1.0
16  16  D.J. Augustin         ORL  7  23.7  2.6  5.7 0.450 1.0 2.6 ⋯ 0.857 1.3
17  17  Luke Babbitt          MIA  7  11.9  1.4  3.4 0.417 0.6 2.3 ⋯ 1.000 0.4
18  18  Dwayne Bacon          CHA  7  23.6  1.7  5.6 0.308 0.7 2.4 ⋯ 0.500 1.0
19  19  Ron Baker             NYK  3  15.9  0.7  2.3 0.286 0.7 1.0 ⋯ 0.556 2.0
20  20  Lonzo Ball            LAL  7  34.0  4.1 12.4 0.333 1.4 5.0 ⋯ 0.556 3.0
21  21  J.J. Barea            DAL  8  23.3  5.1 11.4 0.451 2.4 5.2 ⋯ 0.667 2.5
22  22  Harrison Barnes       DAL  8  35.3  5.9 15.5 0.379 1.5 4.6 ⋯ 0.750 1.6
23  23  Will Barton           DEN  7  27.6  5.4 12.0 0.452 1.9 4.7 ⋯ 0.533 1.4
24  24  Jerryd Bayless        PHI  7  31.4  4.0  8.6 0.467 2.4 5.3 ⋯ 0.800 1.7
25  25  Aron Baynes           BOS  7  19.4  2.6  4.4 0.581 0.0 0.0 ⋯ 0.733 1.0
26  26  Kent Bazemore         ATL  7  29.4  4.3 11.7 0.366 1.0 3.1 ⋯ 0.893 2.7
27  27  Bradley Beal          WAS  6  32.3  7.0 15.5 0.452 1.0 4.0 ⋯ 0.794 2.0
28  28  Malik Beasley         DEN  7   7.0  0.7  1.4 0.500 0.3 0.6 ⋯ 0.000 0.1
29  29  Michael Beasley       NYK  5  12.1  1.8  4.4 0.409 0.0 0.8 ⋯ 0.571 1.2
30  30  Marco Belinelli       PHI  7  27.2  4.9 10.7 0.453 3.1 5.7 ⋯ 1.000 1.7
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮    ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
307 307 Karl-Anthony Towns    MIN  7  34.9 9.3  16.0 0.580 1.3 3.4 ⋯ 0.882 2.0
308 308 P.J. Tucker           HOU  8  30.1 2.0   5.9 0.340 1.6 4.5 ⋯ 0.700 1.5
309 309 Evan Turner           POR  7  27.5 3.7   8.1 0.456 0.7 2.3 ⋯ 0.824 1.7
310 310 Ekpe Udoh             UTA  7  14.2 0.3   1.4 0.200 0.0 0.0 ⋯ 0.571 0.4
311 311 Tyler Ulis            PHX  6  18.3 3.2   7.2 0.442 0.2 1.2 ⋯ 0.778 1.3
312 312 Jonas Valanciunas     TOR  2  22.8 5.5  10.5 0.524 0.0 0.5 ⋯ 0.750 0.5
313 313 Denzel Valentine      CHI  5  27.2 3.4  10.6 0.321 2.2 5.6 ⋯ 1.000 1.4
314 314 Fred VanVleet         TOR  6  11.6 1.0   4.0 0.250 0.3 1.5 ⋯ 1.000 1.0
315 315 Rashad Vaughn         ORL  4  10.9 2.0   4.0 0.500 1.2 2.5 ⋯ 0.000 0.2
316 316 Nikola Vucevic        ORL  7  30.6 8.7  15.7 0.555 1.9 4.6 ⋯ 0.714 1.4
317 317 Dwyane Wade           MIA  6  22.8 3.7   8.5 0.431 0.3 1.0 ⋯ 0.600 2.2
318 318 Dion Waiters          MIA  6  32.7 6.8  15.3 0.446 1.5 5.7 ⋯ 0.667 2.3
319 319 Kemba Walker          CHA  7  34.8 7.4  16.0 0.464 2.1 5.6 ⋯ 0.912 2.6
320 320 John Wall             WAS  6  35.1 7.0  17.8 0.393 1.5 4.5 ⋯ 0.822 4.3
321 321 T.J. Warren           PHX  7  27.4 6.0  12.6 0.477 0.1 1.1 ⋯ 0.760 1.7
322 322 David West            GSW  8  11.6 3.2   4.8 0.684 0.2 0.4 ⋯ 0.667 1.1
323 323 Russell Westbrook     OKC  7  33.2 7.3  15.4 0.472 1.3 3.6 ⋯ 0.619 5.3
324 324 Andrew Wiggins        MIN  7  35.4 6.9  16.0 0.429 1.9 5.1 ⋯ 0.592 2.0
325 325 Damien Wilkins        IND  6   6.9 0.3   1.0 0

The `max()` function was then used to find the maximum value in the PPG (points per game) column, to select this column October(the name of the table)$PPG was done na.rm = TRUE was placed in the function to omit any NA values within the table.


In [14]:
max(October$PPG, na.rm = TRUE)

[1] 33.7

The highest PPG value in October was 33.7

And finally, the row containing the found value using the `max()` function was selected out of the table by using the code: October[October$PPG %in% c("33.7"), ]  where October is the table of interest, `October$PPG` selects the column of interest and %in% c(“33.7”) take the value of interest.

In [15]:
October[October$PPG %in% c("33.7"), ]

#  Player                Team GP MPG  FGM FGA  FG%   3PM 3PA ⋯ FT%   TOV PF 
12 12 Giannis Antetokounmpo MIL  7  37.4 13  20.6 0.632 0.9 2.1 ⋯ 0.774 3.3 3.6
   ORB DRB RPG  APG SPG BPG PPG 
12 1.4 8.9 10.3 5.3 1.7 1   33.7

The player with the most Points Per Game in October is Giannis Antetokounmpo

These steps above were then completed in the exact same format for the next 6 months of the NBS season 

### November

In [16]:
NBA5 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/November")

In [17]:
NBA6 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/November")

In [18]:
NBA7 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/November")

In [19]:
NBA8 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/November")

In [20]:
November1 = NBA5 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [21]:
November2 = NBA6 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [22]:
November3 = NBA7 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [23]:
November4 = NBA8 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [24]:
c = full_join(November1, November2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [25]:
d = full_join(November3, November4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [26]:
November = full_join(c, d)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 2: The data and statistics for all players in the NBA during the month of November in the 2017-2018 season.

In [27]:
November

#   Player                Team GP MPG  FGM FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  12 12.8 1.2  3.2 0.385 0.8 2.4 ⋯ 0.778 0.2
2    2  Quincy Acy            BRK  10 20.6 1.3  5.3 0.245 0.9 3.9 ⋯ 0.818 0.8
3    3  Steven Adams          OKC  10 31.3 4.2  7.6 0.553 0.0 0.0 ⋯ 0.655 2.0
4    4  Bam Adebayo           MIA   8 11.5 1.9  2.2 0.833 0.0 0.0 ⋯ 0.833 0.4
5    5  Arron Afflalo         ORL  12 12.8 0.8  2.3 0.357 0.4 1.2 ⋯ 0.333 0.5
6    6  LaMarcus Aldridge     SAS  14 33.0 8.9 16.7 0.534 0.6 1.5 ⋯ 0.861 1.4
7    7  Jarrett Allen         BRK   7 15.8 1.9  3.4 0.542 0.0 0.0 ⋯ 0.833 0.4
8    8  Tony Allen            NOP  10 15.6 2.6  5.3 0.491 0.2 0.6 ⋯ 0.556 1.0
9    9  Al-Farouq Aminu       POR   2 26.9 2.0  4.5 0.444 2.0 3.5 ⋯ 0.500 1.5
10  10  Justin Anderson       PHI   5 13.3 2.0  4.6 0.435 0.8 2.2 ⋯ 1.000 0.4
11  11  Kyle Anderson         SAS  14 28.5 3.9  6.9 0.557 0.1 0.4 ⋯ 0.811 1.4
12  12  Ryan Anderson         HOU  12 28.8 4.4  8.9 0.495 2.8 6.6 ⋯ 0.833 1.0
13  13  Giannis Antetokounmpo MIL  12 37.1 9.9 20.0 0.496 0.2 1.5 ⋯ 0.748 2.5
14  14  Carmelo Anthony       OKC  12 32.0 6.2 15.9 0.387 2.4 6.3 ⋯ 0.804 1.6
15  15  Ogugua Anunoby        TOR  14 19.3 2.4  4.2 0.576 1.1 2.4 ⋯ 0.562 0.4
16  16  Trevor Ariza          HOU  13 33.5 5.1 10.2 0.496 3.5 7.9 ⋯ 0.833 1.0
17  17  D.J. Augustin         ORL   8 17.0 2.8  5.0 0.550 1.0 2.1 ⋯ 0.733 1.8
18  18  Luke Babbitt          MIA  11 24.4 3.7  7.6 0.488 2.5 5.0 ⋯ 0.625 0.9
19  19  Dwayne Bacon          CHA  13 16.6 1.8  4.5 0.407 0.4 0.9 ⋯ 0.857 0.5
20  20  Lonzo Ball            LAL  14 33.1 3.2 10.6 0.304 1.2 5.0 ⋯ 0.429 2.4
21  21  J.J. Barea            DAL  14 21.5 3.9  9.8 0.401 1.6 4.9 ⋯ 0.815 1.8
22  22  Harrison Barnes       DAL  14 35.5 7.1 15.1 0.474 1.1 3.4 ⋯ 0.867 2.2
23  23  Will Barton           DEN  14 30.6 5.7 11.5 0.497 2.2 4.9 ⋯ 0.735 2.0
24  24  Nicolas Batum         CHA   7 26.3 3.6  9.6 0.373 0.9 4.0 ⋯ 0.938 1.6
25  25  Jerryd Bayless        PHI   8 25.1 2.9  6.6 0.434 0.8 2.9 ⋯ 0.765 1.2
26  26  Aron Baynes           BOS  16 17.4 2.2  4.9 0.443 0.0 0.2 ⋯ 0.600 1.0
27  27  Kent Bazemore         ATL  14 27.9 4.1 10.0 0.407 1.5 4.1 ⋯ 0.723 2.3
28  28  Bradley Beal          WAS  15 35.7 8.6 18.6 0.462 2.8 6.8 ⋯ 0.826 2.9
29  29  Malik Beasley         DEN  12  9.2 1.3  3.2 0.410 0.4 1.5 ⋯ 0.545 0.6
30  30  Michael Beasley       NYK  10 15.6 3.8  7.0 0.543 0.0 0.5 ⋯ 0.615 1.9
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮   ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
352 352 Rashad Vaughn         ORL   6  3.7 0.2  0.8 0.200 0.0 0.3 ⋯ 0.000 0.0
353 353 Noah Vonleh           CHI  15 22.8 2.1  3.9 0.542 0.0 0.0 ⋯ 0.750 0.7
354 354 Nikola Vucevic        ORL  15 29.4 6.0 13.5 0.443 1.4 4.1 ⋯ 0.800 2.0
355 355 Dwyane Wade           MIA  15 23.7 4.6 10.5 0.437 0.5 1.3 ⋯ 0.741 2.0
356 356 Dion Waiters          MIA  13 30.8 5.6 14.9 0.376 2.2 6.7 ⋯ 0.826 2.9
357 357 Kemba Walker          CHA  12 35.4 7.2 16.8 0.431 2.5 7.0 ⋯ 0.803 2.0
358 358 John Wall             WAS  10 34.0 6.9 14.9 0.463 1.0 3.3 ⋯ 0.701 2.2
359 359 T.J. Warren           PHX  16 32.4 8.2 16.4 0.502 0.2 1.6 ⋯ 0.754 1.1
360 360 David West            GSW  12 12.1 2.7  4.1 0.653 0.1 0.2 ⋯ 0.818 1.0
361 361 Russell Westbrook     OKC  13 36.4 8.0 21.1 0.380 2.3 6.7 ⋯ 0.766 4.3
362 362 Okaro White           MIA   4 15.7 1.2  3.5 0.357 0.8 2.5 ⋯ 1.000 0.2
363 363 Isaiah Whitehead      BRK   6 14.0 3.3  6.8 0.488 0.3 1.3 ⋯ 0.600 2.7
364 364 Hassan Whiteside      MIA  14 26.1 5.9 10.1 0.577 0.1 0.1 ⋯ 0.673 1.6
365 365 Andrew Wiggins        MIN  15 37.4 7.2 15.4 0.468 1.1 3.8 ⋯ 0.676 1.9
366 366 Damien Wilkins        IND   6  6.0 1.0  1.7 0.600 0.0 0.3 ⋯ 0.000 0.3
367 367 C.J. Williams         LAC   8  7.6 0.9  1.9 0.467 0.1 0.8 ⋯ 0.500 0.2
368 368 Lou Williams          LAC  14 32.3 7.1 15.3 0.463 2.7 6.7 ⋯ 0.908 3.4
369 369 Marvin Williams       CHA  13 25.7 3.2  7.0 0.462 1.8 3.8 ⋯ 1.000 0.3
370 370 D.J. Wilson           MIL   6  3.4 0.3  0.8 0.400 0.0 0.3 ⋯ 0.000 0.2
371 371 Justise Winslow   

In [28]:
max(November$PPG, na.rm = TRUE)

[1] 34.9

The highest PPG value in the month of November was 34.9

In [29]:
November[November$PPG %in% c("34.9"), ]

#   Player       Team GP MPG  FGM  FGA  FG%   3PM 3PA  ⋯ FT%   TOV PF  ORB
146 146 James Harden HOU  13 35.7 10.2 21.7 0.468 5.2 11.9 ⋯ 0.891 4.4 2.5 0.5
    DRB RPG APG  SPG BPG PPG 
146 4.9 5.5 10.1 1.6 0.5 34.9

The player with the most Points Per Game in November is James Harden

### December

In [30]:
NBA9 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/December")

In [31]:
NBA10 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/December")

In [32]:
NBA11 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/Decemeber")

In [33]:
NBA12 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/Decemeber")

In [34]:
December1 = NBA9 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [35]:
December2 = NBA10 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [36]:
December3 = NBA11 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [37]:
December4 = NBA12 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [38]:
e = full_join(December1, December2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [39]:
f = full_join(December3, December4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [40]:
December = full_join(e, f)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 3: The data and statistics for all players in the NBA during the month of December in the 2017-2018 season

In [41]:
December

#   Player                Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  16 17.2  1.6  4.0 0.406 1.3 2.9 ⋯ 0.800 0.7
2    2  Quincy Acy            BRK  15 18.0  2.1  5.7 0.360 1.5 4.2 ⋯ 0.812 1.0
3    3  Steven Adams          OKC  15 33.6  6.4  9.7 0.658 0.0 0.1 ⋯ 0.642 1.3
4    4  Bam Adebayo           MIA  15 24.5  3.1  5.3 0.582 0.0 0.1 ⋯ 0.721 1.3
5    5  Arron Afflalo         ORL  12 16.4  1.4  3.8 0.378 0.3 1.2 ⋯ 0.917 0.2
6    6  LaMarcus Aldridge     SAS  15 33.8  7.9 17.5 0.450 0.5 1.5 ⋯ 0.863 1.7
7    7  Jarrett Allen         BRK  14 18.5  2.7  5.6 0.487 0.1 0.4 ⋯ 0.900 0.9
8    8  Tony Allen            NOP   5  8.4  1.2  2.6 0.462 0.2 0.2 ⋯ 0.625 0.4
9    9  Al-Farouq Aminu       POR  13 30.1  3.2  8.3 0.389 2.1 5.3 ⋯ 0.375 1.3
10  10  Kyle Anderson         SAS   8 23.3  2.5  4.2 0.588 0.2 0.5 ⋯ 0.824 1.1
11  11  Ryan Anderson         HOU  14 26.4  2.5  6.7 0.372 1.4 4.4 ⋯ 0.765 1.1
12  12  Giannis Antetokounmpo MIL  13 39.3 10.4 19.0 0.547 0.5 1.8 ⋯ 0.802 3.1
13  13  Carmelo Anthony       OKC  17 33.7  5.9 14.5 0.407 1.8 5.0 ⋯ 0.684 1.2
14  14  Ogugua Anunoby        TOR  14 24.1  2.6  5.7 0.463 1.6 3.8 ⋯ 0.667 1.2
15  15  Trevor Ariza          HOU  14 38.8  4.5 10.1 0.447 2.8 7.1 ⋯ 0.875 0.6
16  16  Jamel Artis           ORL   5 11.0  0.8  3.4 0.235 0.0 1.0 ⋯ 0.333 0.0
17  17  Omer Asik             CHI   9  7.5  0.2  0.8 0.286 0.0 0.0 ⋯ 0.500 0.2
18  18  D.J. Augustin         ORL  15 21.6  2.4  6.3 0.379 1.5 3.7 ⋯ 0.919 1.5
19  19  Luke Babbitt          MIA  10 15.2  1.7  3.7 0.459 1.0 2.2 ⋯ 0.778 0.1
20  20  Dwayne Bacon          CHA   8  8.1  0.6  1.5 0.417 0.0 0.1 ⋯ 0.667 0.2
21  21  Ron Baker             NYK  12 15.9  0.9  2.6 0.355 0.6 1.5 ⋯ 0.867 0.7
22  22  Lonzo Ball            LAL  10 34.8  4.6 10.9 0.422 2.0 5.3 ⋯ 0.500 3.4
23  23  J.J. Barea            DAL  16 24.1  4.7  9.8 0.481 1.8 4.2 ⋯ 0.727 2.1
24  24  Harrison Barnes       DAL  16 33.8  7.2 15.5 0.464 1.3 3.6 ⋯ 0.787 1.2
25  25  Will Barton           DEN  14 33.8  5.3 13.0 0.407 1.5 5.5 ⋯ 0.800 1.4
26  26  Nicolas Batum         CHA  15 33.1  4.1  9.7 0.418 1.2 3.8 ⋯ 0.722 2.2
27  27  Jerryd Bayless        PHI  15 23.3  2.8  7.5 0.375 1.7 4.5 ⋯ 0.824 1.1
28  28  Aron Baynes           BOS  17 18.7  2.5  5.1 0.500 0.0 0.2 ⋯ 0.889 0.9
29  29  Kent Bazemore         ATL  15 28.5  4.1  9.6 0.431 1.8 4.3 ⋯ 0.861 2.5
30  30  Bradley Beal          WAS  16 36.2  9.0 20.8 0.434 2.5 7.2 ⋯ 0.716 1.3
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮    ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
326 326 Evan Turner           POR  79 25.7 3.3   7.3 0.447 0.5 1.7 ⋯ 0.850 1.3
327 327 Myles Turner          IND  65 28.3 4.7   9.8 0.479 0.9 2.4 ⋯ 0.777 1.5
328 328 Ekpe Udoh             UTA  63 12.8 1.0   1.9 0.500 0.0 0.0 ⋯ 0.750 0.3
329 329 Tyler Ulis            PHX  71 23.4 3.0   7.8 0.388 0.6 2.1 ⋯ 0.832 1.8
330 330 Jonas Valanciunas     TOR  77 22.4 5.1   8.9 0.568 0.4 1.0 ⋯ 0.806 1.5
331 331 Denzel Valentine      CHI  77 27.2 3.9   9.4 0.417 1.9 4.8 ⋯ 0.745 1.3
332 332 Fred VanVleet         TOR  76 20.0 3.1   7.2 0.426 1.4 3.4 ⋯ 0.832 1.0
333 333 Noah Vonleh           CHI  54 16.2 1.9   4.3 0.444 0.4 1.2 ⋯ 0.621 0.6
334 334 Nikola Vucevic        ORL  57 29.5 7.0  14.7 0.475 1.1 3.6 ⋯ 0.819 1.9
335 335 Dwyane Wade           MIA  67 22.9 4.5  10.2 0.438 0.5 1.7 ⋯ 0.714 2.1
336 336 Dion Waiters          MIA  30 30.6 5.4  13.7 0.398 1.7 5.7 ⋯ 0.739 2.3
337 337 Kemba Walker          CHA  80 34.2 7.3  17.0 0.431 2.9 7.5 ⋯ 0.864 2.2
338 338 John Wall             WAS  41 34.4 6.8  16.3 0.420 1.5 4.1 ⋯ 0.726 3.9
339 339 Tyrone Wallace        LAC  30 28.4 3.7   8.2 0.445 0.3 1.3 ⋯ 0.782 1.5
340 340 T.J. Warren           PHX  65 33.0 8.1  16.4 0.498 0.3 1.4 ⋯ 0.757 1.3
341 341 David West            GSW  73 13.7 3.0   5.2 0.571 0.0 0.1 ⋯ 0.759 1.1
342 342 Russell Westbrook     OKC  80 36.4 9.5  21.1 0.449 1.2 4.1 ⋯ 0.737 4.8
343 343 Hassan Whiteside      MIA  54 25.3 5.8  10.7 0.540 0.0 0.0 ⋯ 0.703 1.7
344 344 Andrew Wiggins        MIN  82 36.3 6.9  15.9 0

In [42]:
max(December$PPG, na.rm = TRUE)

[1] 33.6

The highest PPG value in the month of December was 33.6 

In [43]:
December[December$PPG %in% c("33.6"), ]

#   Player       Team GP MPG  FGM FGA  FG%  3PM 3PA ⋯ FT%   TOV PF  ORB DRB
145 145 James Harden HOU  14 36.9 9.6 21.3 0.45 3.6 9.9 ⋯ 0.868 4.1 2.8 0.3 4.4
    RPG APG SPG BPG PPG 
145 4.7 8.1 2   0.6 33.6

The player with the most Points Per Game in December is James Harden 

### January

In [44]:
NBA13 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/January")

In [45]:
NBA14 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/January")

In [46]:
NBA15 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/January")

In [47]:
NBA16 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/January")

In [48]:
January1 = NBA13 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [49]:
January2 = NBA14 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [50]:
January3 = NBA15 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [51]:
January4 = NBA16 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [52]:
g = full_join(January1, January2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [53]:
h = full_join(January3, January4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [54]:
January = full_join(g, h)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 4: The data and statistics for all players in the NBA during the month of January in the 2017-2018 season.

In [55]:
January

#   Player                Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  10  8.3 1.0   2.7 0.370 0.7 2.0 ⋯ 1.000 0.1
2    2  Quincy Acy            BRK  16 23.1 1.8   5.2 0.333 1.4 4.2 ⋯ 0.750 1.1
3    3  Steven Adams          OKC  14 32.2 6.6  10.1 0.648 0.0 0.1 ⋯ 0.449 1.7
4    4  Bam Adebayo           MIA  15 20.0 2.5   5.1 0.500 0.0 0.2 ⋯ 0.774 1.1
5    5  Arron Afflalo         ORL   9 10.8 1.6   3.1 0.500 0.8 1.2 ⋯ 0.000 0.4
6    6  LaMarcus Aldridge     SAS  14 35.5 9.4  19.1 0.494 0.4 1.4 ⋯ 0.831 1.7
7    7  Jarrett Allen         BRK  16 18.2 3.6   4.9 0.722 0.1 0.1 ⋯ 0.714 1.5
8    8  Al-Farouq Aminu       POR  16 30.1 4.1   9.9 0.415 2.1 5.4 ⋯ 0.800 0.8
9    9  Justin Anderson       PHI  10 16.4 2.5   5.6 0.446 0.6 2.4 ⋯ 0.909 0.3
10  10  Kyle Anderson         SAS  16 29.3 2.8   5.6 0.500 0.4 0.9 ⋯ 0.850 1.7
11  11  Ryan Anderson         HOU  14 27.8 2.9   7.0 0.408 1.4 4.4 ⋯ 0.786 0.6
12  12  Giannis Antetokounmpo MIL  13 35.2 9.3  16.9 0.550 0.6 1.5 ⋯ 0.754 2.4
13  13  Carmelo Anthony       OKC  14 31.6 6.9  16.0 0.429 2.2 6.6 ⋯ 0.789 0.9
14  14  Ogugua Anunoby        TOR  15 22.0 1.7   4.3 0.391 0.5 2.4 ⋯ 0.812 0.5
15  15  Ryan Arcidiacono      CHI   6 11.9 0.5   0.8 0.600 0.3 0.7 ⋯ 1.000 0.2
16  16  Trevor Ariza          HOU  11 31.4 3.5   9.4 0.379 2.6 7.1 ⋯ 0.867 1.1
17  17  Omer Asik             CHI   4 10.6 0.5   1.0 0.500 0.0 0.0 ⋯ 0.167 0.8
18  18  D.J. Augustin         ORL  13 18.6 2.8   6.3 0.439 1.4 3.2 ⋯ 0.769 1.1
19  19  Luke Babbitt          MIA   7  7.5 1.4   2.6 0.556 0.9 1.9 ⋯ 1.000 0.1
20  20  Ron Baker             NYK  12 11.3 0.5   1.6 0.316 0.2 1.0 ⋯ 1.000 0.2
21  21  Lonzo Ball            LAL   5 34.5 4.4  11.0 0.400 2.4 7.4 ⋯ 0.000 1.8
22  22  J.J. Barea            DAL  10 22.6 3.0   8.8 0.341 0.7 3.3 ⋯ 0.815 1.3
23  23  Harrison Barnes       DAL  14 34.8 7.0  15.6 0.450 1.5 4.3 ⋯ 0.848 1.1
24  24  Will Barton           DEN  15 33.0 4.6  11.9 0.385 1.5 4.7 ⋯ 0.833 2.5
25  25  Nicolas Batum         CHA  14 32.2 4.7  11.2 0.420 1.5 4.6 ⋯ 0.775 2.5
26  26  Jerryd Bayless        PHI   5 19.8 2.0   4.2 0.476 0.6 2.0 ⋯ 0.000 0.6
27  27  Aron Baynes           BOS  12 17.4 2.3   6.0 0.389 0.0 0.2 ⋯ 0.667 1.0
28  28  Kent Bazemore         ATL  15 27.4 5.3  11.3 0.465 1.9 4.9 ⋯ 0.755 2.5
29  29  Bradley Beal          WAS  13 38.2 8.8  17.5 0.504 2.7 6.6 ⋯ 0.785 3.0
30  30  Malik Beasley         DEN  12 12.3 1.2   3.3 0.375 0.4 1.7 ⋯ 0.600 0.3
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮    ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
349 349 Fred VanVleet         TOR  13 19.2  4.2  7.9 0.524 1.9 3.8 ⋯ 0.880 0.8
350 350 Noah Vonleh           CHI   7  5.0  0.7  2.3 0.312 0.3 0.7 ⋯ 1.000 0.1
351 351 Dwyane Wade           MIA  11 22.1  4.5  9.2 0.485 0.2 1.1 ⋯ 0.727 1.9
352 352 Kemba Walker          CHA  14 35.1  8.1 18.9 0.430 3.4 8.2 ⋯ 0.842 2.4
353 353 John Wall             WAS  11 36.8  7.9 18.6 0.424 1.7 4.0 ⋯ 0.762 4.9
354 354 Tyrone Wallace        LAC  13 32.7  4.5  9.2 0.483 0.5 1.5 ⋯ 0.739 1.5
355 355 Derrick Walton, Jr.   MIA   3 12.3  0.7  2.0 0.333 0.7 2.0 ⋯ 0.000 0.0
356 356 T.J. Warren           PHX  11 34.2  9.5 16.6 0.568 0.5 1.5 ⋯ 0.677 1.3
357 357 James Webb III        BRK   3 12.6  0.0  2.7 0.000 0.0 2.3 ⋯ 0.000 0.0
358 358 Briante Weber         MEM   5  9.7  0.8  2.4 0.333 0.2 1.0 ⋯ 1.000 0.4
359 359 David West            GSW  14 14.1  3.3  5.8 0.568 0.0 0.0 ⋯ 0.812 1.3
360 360 Russell Westbrook     OKC  14 34.9 10.9 22.4 0.484 0.9 2.6 ⋯ 0.685 4.8
361 361 Derrick White         SAS   6  5.2  0.3  0.7 0.500 0.2 0.5 ⋯ 0.750 0.3
362 362 Hassan Whiteside      MIA  15 25.6  6.0 10.7 0.562 0.0 0.0 ⋯ 0.809 2.3
363 363 Andrew Wiggins        MIN  17 35.6  7.9 17.4 0.453 1.2 4.1 ⋯ 0.696 1.5
364 364 Damien Wilkins        IND   2 13.4  1.5  4.5 0.333 1.5 3.0 ⋯ 1.000 0.0
365 365 C.J. Williams         LAC   5 28.3  4.6  8.4 0.548 1.6 3.8 ⋯ 0.750 1.0
366 366 Lou Williams          LAC  15 35.4  9.0 20.4 0.441 3.3 9.1 ⋯ 0.904 3.6
367 367 Marvin Williams       CHA  13 26.0  3.9  7.4 0

In [56]:
max(January$PPG, na.rm = TRUE)

[1] 29.5

The highest PPG value in the month of January was 29.5

In [57]:
January[January$PPG %in% c("29.5"), ]

#  Player        Team GP MPG  FGM FGA  FG%   3PM 3PA  ⋯ FT%   TOV PF  ORB
86 86 Stephen Curry GSW  12 34.3 9.1 17.7 0.514 5.2 11.2 ⋯ 0.871 3.1 2.2 0.6
   DRB RPG APG SPG BPG PPG 
86 4.5 5.1 6.7 1.5 0   29.5

The player with the most Points Per Game in January is Stephen Curry 

### February

In [58]:
NBA17 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/February")

In [59]:
NBA18 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/February")

In [60]:
NBA19 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/February")

In [61]:
NBA20 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/February")

In [62]:
February1 = NBA17 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [63]:
February2 = NBA18 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [64]:
February3 = NBA19 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [65]:
February4 = NBA20 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [66]:
i = full_join(February1, February2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [67]:
j = full_join(February3, February4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [68]:
February = full_join(i, j)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 5: The data and statistics for all players in the NBA during the month of February in the 2017-2018 season

In [69]:
February

#   Player                Team GP MPG  FGM FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  12 22.3 2.3  6.2 0.373 1.7 4.9 ⋯ 0.889 0.4
2    2  Quincy Acy            BRK   8 23.1 1.5  5.1 0.293 1.1 4.4 ⋯ 1.000 1.0
3    3  Steven Adams          OKC  12 34.6 5.8  9.5 0.614 0.0 0.0 ⋯ 0.544 1.6
4    4  Bam Adebayo           MIA  10 21.5 2.8  5.6 0.500 0.0 0.2 ⋯ 0.700 0.7
5    5  Arron Afflalo         ORL   8 12.5 1.4  3.5 0.393 0.8 1.8 ⋯ 1.000 0.6
6    6  LaMarcus Aldridge     SAS   7 30.2 9.3 16.1 0.575 0.1 0.4 ⋯ 0.821 0.9
7    7  Jarrett Allen         BRK  10 23.9 5.3  8.6 0.616 0.0 0.2 ⋯ 0.821 0.8
8    8  Al-Farouq Aminu       POR  10 30.6 2.8  7.3 0.384 1.6 4.2 ⋯ 0.786 0.9
9    9  Justin Anderson       PHI   6 12.6 2.7  4.8 0.552 1.2 2.2 ⋯ 0.286 0.3
10  10  Kyle Anderson         SAS   9 26.9 3.4  6.7 0.517 0.0 0.9 ⋯ 0.542 1.2
11  11  Ryan Anderson         HOU   9 23.3 3.4  7.1 0.484 2.3 5.0 ⋯ 0.727 0.1
12  12  Giannis Antetokounmpo MIL  12 36.7 8.8 18.5 0.477 0.5 2.0 ⋯ 0.679 3.2
13  13  Carmelo Anthony       OKC  10 30.9 5.4 14.8 0.365 2.0 5.8 ⋯ 0.750 1.1
14  14  Ogugua Anunoby        TOR  11 16.9 1.9  4.4 0.438 0.7 2.3 ⋯ 0.625 0.5
15  15  Ryan Arcidiacono      CHI   6  7.1 0.3  0.8 0.400 0.2 0.7 ⋯ 0.000 0.3
16  16  Trevor Ariza          HOU   4 31.1 4.0  9.0 0.444 3.0 6.2 ⋯ 0.500 0.0
17  17  Darrell Arthur        DEN   7  8.6 1.9  2.9 0.650 0.6 1.1 ⋯ 0.500 1.0
18  18  D.J. Augustin         ORL  11 27.0 3.5  8.4 0.413 1.9 4.6 ⋯ 0.935 1.6
19  19  Luke Babbitt          MIA   5  7.8 0.8  2.4 0.333 0.8 2.4 ⋯ 0.000 0.4
20  20  Dwayne Bacon          CHA   8  3.0 0.4  0.9 0.429 0.0 0.1 ⋯ 0.000 0.1
21  21  Lonzo Ball            LAL   2 20.7 3.5  6.0 0.583 3.0 4.5 ⋯ 1.000 2.0
22  22  J.J. Barea            DAL   9 26.9 4.9 10.7 0.458 1.9 5.4 ⋯ 0.875 2.1
23  23  Harrison Barnes       DAL   8 31.8 6.2 14.0 0.446 2.0 5.2 ⋯ 0.900 1.5
24  24  Will Barton           DEN  10 36.5 7.1 13.2 0.538 2.4 5.5 ⋯ 0.955 1.5
25  25  Nicolas Batum         CHA  12 31.4 4.8 11.7 0.407 2.3 6.2 ⋯ 0.842 1.3
26  26  Jerryd Bayless        PHI   4 13.7 1.5  4.0 0.375 0.8 2.0 ⋯ 0.000 1.0
27  27  Aron Baynes           BOS  10 16.9 2.9  5.3 0.547 0.0 0.1 ⋯ 0.778 0.6
28  28  Kent Bazemore         ATL  10 26.1 3.6 10.1 0.356 1.5 3.9 ⋯ 0.806 2.2
29  29  Bradley Beal          WAS  12 37.7 8.5 18.3 0.464 2.3 6.5 ⋯ 0.879 3.8
30  30  Malik Beasley         DEN   9  7.6 1.0  2.4 0.409 0.3 1.0 ⋯ 0.667 0.3
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮   ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
338 338 Jonas Valanciunas     TOR  11 24.8 5.0  8.8 0.567 0.9 1.8 ⋯ 0.556 1.4
339 339 Denzel Valentine      CHI  10 24.4 3.6 10.2 0.353 1.2 4.4 ⋯ 0.900 1.3
340 340 Fred VanVleet         TOR  10 23.3 3.8  8.6 0.442 1.9 4.1 ⋯ 0.917 1.2
341 341 Rashad Vaughn         ORL   7  4.9 0.3  0.9 0.333 0.1 0.3 ⋯ 0.000 0.4
342 342 Noah Vonleh           CHI   7 11.5 1.6  2.9 0.550 0.4 1.1 ⋯ 0.500 0.6
343 343 Nikola Vucevic        ORL   4 29.0 6.8 12.5 0.540 1.5 3.8 ⋯ 0.600 2.2
344 344 Dwyane Wade           MIA   8 22.3 4.6 11.0 0.420 0.6 2.0 ⋯ 0.706 2.6
345 345 Kemba Walker          CHA  12 34.0 8.5 18.1 0.470 4.0 8.2 ⋯ 0.917 2.3
346 346 Tyrone Wallace        LAC   8 19.5 2.4  5.9 0.404 0.1 0.8 ⋯ 0.889 0.6
347 347 T.J. Warren           PHX  10 33.8 7.2 15.4 0.468 0.2 0.5 ⋯ 0.750 0.4
348 348 James Webb III        BRK   6 11.7 0.8  2.2 0.385 0.7 1.7 ⋯ 0.000 0.5
349 349 David West            GSW   9 14.9 3.3  5.0 0.667 0.0 0.0 ⋯ 0.900 1.2
350 350 Russell Westbrook     OKC  10 37.4 7.7 18.9 0.407 0.7 4.5 ⋯ 0.894 5.7
351 351 Andrew White          ATL   3 12.2 2.7  6.0 0.444 1.0 3.7 ⋯ 0.000 0.3
352 352 Derrick White         SAS   3 14.0 2.3  4.7 0.500 1.0 1.7 ⋯ 0.750 0.3
353 353 Hassan Whiteside      MIA   9 27.1 6.0 11.7 0.514 0.0 0.0 ⋯ 0.469 1.6
354 354 Andrew Wiggins        MIN  10 36.1 6.5 13.8 0.471 1.2 3.9 ⋯ 0.636 1.9
355 355 Lou Williams          LAC  10 32.9 7.3 16.4 0.445 1.8 6.4 ⋯ 0.871 3.0
356 356 Marvin Williams       CHA   9 25.3 3.1  7.6 0.412 1.9 4.3 ⋯ 0.706 1.6
357 357 Troy Williams     

In [70]:
max(February$PPG, na.rm = TRUE)

[1] 35

The highest PPG value in the month of February was 35

In [71]:
February[February$PPG %in% c("35"), ]

#  Player        Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV PF  ORB
84 84 Anthony Davis NOP  11 37.8 13.1 25.8 0.507 1.2 3.5 ⋯ 0.832 2.5 2.3 3.4
   DRB RPG APG SPG BPG PPG
84 9.6 13  1.5 2.5 2.2 35

The player with the most Points Per Game in February is Anthony Davis

### March

In [72]:
NBA21 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/March")

In [73]:
NBA22 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/March")

In [74]:
NBA23 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/March")

In [75]:
NBA24 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/March")

In [76]:
March1 = NBA21 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [77]:
March2 = NBA22 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [78]:
March3 = NBA23 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [79]:
March4 = NBA24 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [80]:
k = full_join(March1, March2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [81]:
l = full_join(March3, March4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [82]:
March = full_join(k, l)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 6: The data and statistics for all players in the NBA during the month of March in the 2017-2018 season

In [83]:
March

#   Player                Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  14  9.9  1.4  2.7 0.500 1.2 2.4 ⋯ 0.500 0.1
2    2  Quincy Acy            BRK   9 13.8  2.0  4.6 0.439 1.7 3.9 ⋯ 1.000 0.1
3    3  Steven Adams          OKC  13 32.7  5.9  9.8 0.602 0.0 0.0 ⋯ 0.525 1.6
4    4  Bam Adebayo           MIA  13 19.2  2.3  5.7 0.405 0.0 0.1 ⋯ 0.593 1.0
5    5  Arron Afflalo         ORL   6 13.8  1.3  3.2 0.421 0.5 1.3 ⋯ 1.000 0.5
6    6  LaMarcus Aldridge     SAS  12 32.0 10.2 19.1 0.537 0.2 0.8 ⋯ 0.812 1.8
7    7  Jarrett Allen         BRK  13 23.2  3.3  5.8 0.566 0.0 0.3 ⋯ 0.833 1.9
8    8  Kadeem Allen          BOS   5  2.4  0.0  0.0 0.000 0.0 0.0 ⋯ 0.000 0.2
9    9  Al-Farouq Aminu       POR  15 30.0  3.3  8.7 0.374 1.7 5.3 ⋯ 0.714 1.4
10  10  Justin Anderson       PHI   7 14.1  2.1  5.9 0.366 0.9 3.3 ⋯ 1.000 0.7
11  11  Kyle Anderson         SAS  14 23.9  2.9  5.1 0.577 0.4 0.9 ⋯ 0.609 1.4
12  12  Ryan Anderson         HOU   7 18.9  2.1  5.0 0.429 1.6 3.6 ⋯ 0.714 0.3
13  13  Giannis Antetokounmpo MIL  14 36.6 10.0 18.5 0.541 0.9 2.2 ⋯ 0.773 3.2
14  14  Carmelo Anthony       OKC  13 31.3  4.9 12.7 0.388 2.5 6.2 ⋯ 0.706 1.3
15  15  Ogugua Anunoby        TOR   8 16.1  2.0  3.6 0.552 0.6 1.4 ⋯ 0.167 0.8
16  16  Ryan Arcidiacono      CHI   5 10.0  1.0  1.8 0.556 0.6 1.4 ⋯ 1.000 0.2
17  17  Trevor Ariza          HOU  15 32.6  4.1 10.5 0.386 2.3 6.8 ⋯ 0.870 0.7
18  18  Jamel Artis           ORL   4 14.6  1.0  2.5 0.400 0.5 1.2 ⋯ 1.000 0.2
19  19  D.J. Augustin         ORL  14 31.4  4.6  9.6 0.485 2.1 4.7 ⋯ 0.870 2.4
20  20  Luke Babbitt          MIA   8 11.0  1.0  3.8 0.267 1.0 3.5 ⋯ 0.000 0.2
21  21  Dwayne Bacon          CHA  13 17.6  2.2  5.4 0.400 0.1 0.8 ⋯ 0.909 0.5
22  22  Lonzo Ball            LAL  14 36.9  3.9 10.9 0.353 1.8 6.5 ⋯ 0.316 2.4
23  23  J.J. Barea            DAL  10 23.2  5.0 10.8 0.463 1.8 4.7 ⋯ 0.833 2.9
24  24  Harrison Barnes       DAL  13 33.0  7.5 17.2 0.433 1.8 4.5 ⋯ 0.810 1.5
25  25  Will Barton           DEN  15 33.2  5.8 13.2 0.439 2.1 5.5 ⋯ 0.902 2.1
26  26  Nicolas Batum         CHA  11 32.7  4.0  9.1 0.440 1.2 2.9 ⋯ 0.885 1.9
27  27  Aron Baynes           BOS  13 19.7  2.6  5.6 0.466 0.2 0.4 ⋯ 0.875 1.3
28  28  Kent Bazemore         ATL   4 23.3  5.0  9.8 0.513 2.2 4.0 ⋯ 0.800 2.2
29  29  Bradley Beal          WAS  14 35.9  7.6 16.4 0.465 2.6 6.4 ⋯ 0.750 3.1
30  30  Malik Beasley         DEN   8  6.7  1.0  2.1 0.471 0.8 1.4 ⋯ 1.000 0.2
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮    ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
370 370 Fred VanVleet         TOR  14 24.8  3.9  9.1 0.433 2.1 5.1 ⋯ 0.759 1.1
371 371 Noah Vonleh           CHI  14 22.0  3.2  8.3 0.388 1.0 3.6 ⋯ 0.526 0.6
372 372 Nikola Vucevic        ORL  14 31.0  7.4 16.3 0.452 0.7 3.1 ⋯ 0.833 2.2
373 373 Dwyane Wade           MIA  11 22.5  5.2 12.2 0.425 0.5 2.0 ⋯ 0.778 2.1
374 374 Kemba Walker          CHA  15 34.8  7.1 16.6 0.430 3.1 8.1 ⋯ 0.895 2.3
375 375 Tyrone Wallace        LAC   3 27.6  3.3  9.3 0.357 0.3 1.0 ⋯ 0.833 2.0
376 376 T.J. Warren           PHX   6 36.0  8.8 17.7 0.500 0.5 1.5 ⋯ 0.846 1.7
377 377 Travis Wear           LAL  13 13.3  1.7  4.4 0.386 1.2 3.2 ⋯ 1.000 0.3
378 378 Briante Weber         MEM   5 23.8  2.0  4.2 0.476 0.0 0.4 ⋯ 0.667 1.6
379 379 David West            GSW   9 14.5  2.9  6.7 0.433 0.0 0.1 ⋯ 0.833 1.6
380 380 Russell Westbrook     OKC  14 36.8 10.6 21.2 0.498 1.1 2.9 ⋯ 0.721 4.6
381 381 Andrew White          ATL  11 14.8  1.5  4.6 0.333 1.4 3.1 ⋯ 0.333 0.5
382 382 Hassan Whiteside      MIA   7 22.5  4.9  9.4 0.515 0.0 0.0 ⋯ 0.812 2.3
383 383 Andrew Wiggins        MIN  13 37.2  7.2 16.6 0.435 1.5 4.1 ⋯ 0.685 2.2
384 384 Alan Williams         PHX   3 16.6  1.3  3.7 0.364 0.0 0.0 ⋯ 0.571 1.7
385 385 C.J. Williams         LAC   3 16.2  2.0  5.0 0.400 0.3 2.0 ⋯ 1.000 1.0
386 386 Lou Williams          LAC  16 33.7  7.4 17.8 0.414 1.4 4.8 ⋯ 0.830 3.1
387 387 Marvin Williams       CHA  15 25.8  2.8  7.1 0.393 0.9 3.8 ⋯ 0.774 0.8
388 388 Troy Williams         NYK  14 18.4  2.9  6.1 0

In [84]:
max(March$PPG, na.rm = TRUE)

[1] 30.9

The highest PPG value in the month of March was 30.9

In [85]:
March[March$PPG %in% c("30.9"), ]

#   Player       Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV PF  ORB
186 186 LeBron James CLE  16 37.3 11.8 21.3 0.551 2   5.5 ⋯ 0.707 3.7 1.9 1.2
    DRB RPG APG SPG BPG PPG 
186 8.3 9.6 9.5 1.3 0.8 30.9

The player with the most Points Per Game in March is LeBron James 

### April

In [86]:
NBA25 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/1/April")

In [87]:
NBA26 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/2/April")

In [88]:
NBA27 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/3/April")

In [89]:
NBA28 = read_html("https://basketball.realgm.com/nba/stats/2018/Averages/Qualified/player/All/asc/4/April")

In [90]:
April1 = NBA25 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [91]:
April2 = NBA26 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [92]:
April3 = NBA27 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [93]:
April4 = NBA28 %>% html_nodes('table')%>% .[[1]] %>%
  html_table()

In [94]:
m = full_join(April1, April2)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [95]:
n = full_join(April3, April4)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


In [96]:
April = full_join(m, n)

Joining, by = c("#", "Player", "Team", "GP", "MPG", "FGM", "FGA", "FG%", "3PM", "3PA", "3P%", "FTM", "FTA", "FT%", "TOV", "PF", "ORB", "DRB", "RPG", "APG", "SPG", "BPG", "PPG")


Table 7: The data and statistics for all players in the NBA during the month of April in the 2017-2018 season

In [97]:
April

#   Player                Team GP MPG  FGM  FGA  FG%   3PM 3PA ⋯ FT%   TOV
1    1  Alex Abrines          OKC  4  20.1  1.5  4.2 0.353 1.0 3.8 ⋯ 0.800 0.0
2    2  Quincy Acy            BRK  6  19.2  2.7  6.2 0.432 2.2 5.2 ⋯ 0.500 0.3
3    3  Steven Adams          OKC  5  33.2  5.2  8.0 0.650 0.0 0.0 ⋯ 0.500 2.8
4    4  Bam Adebayo           MIA  4  15.9  2.0  4.2 0.471 0.0 0.0 ⋯ 1.000 1.2
5    5  Arron Afflalo         ORL  2  12.4  2.5  4.5 0.556 1.0 2.5 ⋯ 0.000 0.0
6    6  LaMarcus Aldridge     SAS  6  35.0 10.0 18.8 0.531 0.0 0.8 ⋯ 0.909 0.8
7    7  Jarrett Allen         BRK  6  24.9  3.5  5.3 0.656 0.3 0.3 ⋯ 0.583 1.2
8    8  Kadeem Allen          BOS  6  11.3  0.8  2.7 0.312 0.0 1.3 ⋯ 0.800 0.8
9    9  Al-Farouq Aminu       POR  6  30.0  2.8  8.0 0.354 1.3 5.0 ⋯ 0.800 1.7
10  10  Justin Anderson       PHI  5  10.9  2.6  5.6 0.464 1.2 3.4 ⋯ 0.750 0.4
11  11  Kyle Anderson         SAS  6  26.7  3.0  5.7 0.529 0.5 1.2 ⋯ 0.636 0.3
12  12  Ryan Anderson         HOU  2  11.8  0.0  1.0 0.000 0.0 0.5 ⋯ 0.667 0.0
13  13  Giannis Antetokounmpo MIL  4  31.8  7.5 17.5 0.429 0.2 2.2 ⋯ 0.789 3.8
14  14  Carmelo Anthony       OKC  5  30.5  5.0 13.4 0.373 1.8 6.6 ⋯ 0.800 1.2
15  15  Ogugua Anunoby        TOR  6  20.7  2.8  5.0 0.567 1.5 3.2 ⋯ 0.600 0.5
16  16  Ryan Arcidiacono      CHI  6  22.4  1.2  3.3 0.350 0.5 2.3 ⋯ 0.500 1.5
17  17  Trevor Ariza          HOU  4  29.4  3.0  8.8 0.343 1.0 5.5 ⋯ 0.500 0.5
18  18  Jamel Artis           ORL  6  27.5  3.8  8.7 0.442 1.0 3.2 ⋯ 0.571 1.2
19  19  Omer Asik             CHI  4  15.3  0.5  1.5 0.333 0.0 0.0 ⋯ 0.000 1.0
20  20  D.J. Augustin         ORL  7  22.4  4.1  8.1 0.509 1.3 3.1 ⋯ 0.923 1.7
21  21  Luke Babbitt          MIA  2  15.8  0.5  5.0 0.100 0.5 5.0 ⋯ 0.000 0.0
22  22  Wade Baldwin IV       POR  4  13.9  2.2  3.5 0.643 0.8 1.0 ⋯ 1.000 0.8
23  23  J.J. Barea            DAL  2  15.3  4.0  7.0 0.571 1.5 2.5 ⋯ 0.000 1.5
24  24  Harrison Barnes       DAL  4  36.0  7.5 17.8 0.423 2.5 7.5 ⋯ 0.812 1.5
25  25  Will Barton           DEN  6  38.3  7.7 15.3 0.500 2.3 6.2 ⋯ 0.786 1.5
26  26  Nicolas Batum         CHA  5  22.6  3.4  8.0 0.425 1.2 3.6 ⋯ 0.909 3.0
27  27  Aron Baynes           BOS  6  19.9  3.8  8.7 0.442 0.2 1.2 ⋯ 1.000 1.0
28  28  Bradley Beal          WAS  6  35.9  7.3 17.0 0.431 2.0 6.0 ⋯ 0.833 1.8
29  29  Michael Beasley       NYK  4  17.9  4.5  9.5 0.474 0.5 2.0 ⋯ 0.778 1.5
30  30  Marco Belinelli       PHI  7  28.3  6.6 12.7 0.517 3.1 7.3 ⋯ 0.833 2.6
⋮   ⋮   ⋮                     ⋮    ⋮  ⋮    ⋮    ⋮    ⋮     ⋮   ⋮   ⋱ ⋮     ⋮  
320 320 P.J. Tucker           HOU  6  28.0 2.2   6.3 0.342 1.7 5.3 ⋯ 0.000 1.2
321 321 Evan Turner           POR  6  28.5 4.0   7.5 0.533 0.8 2.5 ⋯ 1.000 1.3
322 322 Myles Turner          IND  6  23.8 3.0   8.0 0.375 0.5 2.3 ⋯ 0.857 1.3
323 323 Ekpe Udoh             UTA  5   6.1 0.4   1.0 0.400 0.0 0.0 ⋯ 0.500 0.0
324 324 Tyler Ulis            PHX  5  34.7 5.2  12.8 0.406 1.0 3.2 ⋯ 0.929 3.0
325 325 Jonas Valanciunas     TOR  5  21.3 5.8  10.4 0.558 0.2 1.2 ⋯ 0.875 1.2
326 326 Fred VanVleet         TOR  5  17.7 2.2   6.4 0.344 1.4 3.4 ⋯ 0.800 1.0
327 327 Noah Vonleh           CHI  2  12.4 1.0   2.5 0.400 0.5 1.0 ⋯ 0.000 1.5
328 328 Nikola Vucevic        ORL  5  23.1 4.2  11.4 0.368 0.0 1.2 ⋯ 0.889 1.0
329 329 Dwyane Wade           MIA  4  21.0 3.8  10.5 0.357 0.2 2.2 ⋯ 0.500 2.0
330 330 Kemba Walker          CHA  5  26.0 3.8   9.4 0.404 1.2 4.8 ⋯ 0.938 2.4
331 331 John Wall             WAS  3  35.8 7.3  15.3 0.478 2.3 4.3 ⋯ 0.789 7.0
332 332 Tyrone Wallace        LAC  6  31.3 3.8   8.7 0.442 0.2 1.8 ⋯ 0.750 2.5
333 333 Travis Wear           LAL  4  13.9 0.8   3.8 0.200 0.2 1.5 ⋯ 0.000 0.5
334 334 David West            GSW  6  12.9 1.7   4.0 0.417 0.0 0.0 ⋯ 0.800 0.7
335 335 Russell Westbrook     OKC  5  38.0 8.8  20.2 0.436 1.2 3.6 ⋯ 0.784 5.6
336 336 Hassan Whiteside      MIA  5  21.9 5.6  11.6 0.483 0.0 0.0 ⋯ 0.750 0.4
337 337 Andrew Wiggins        MIN  5  36.5 5.8  15.4 0.377 1.2 3.2 ⋯ 0.615 0.8
338 338 Alan Williams         PHX  2  10.2 1.5   3.5 0

In [98]:
max(April$PPG, na.rm = TRUE)

[1] 30

The highest PPG value in the month of April was 30

In [99]:
April[April$PPG %in% c("30"), ]

#   Player         Team GP MPG FGM FGA  FG%   3PM 3PA ⋯ FT%   TOV PF ORB
201 201 Damian Lillard POR  5  36  9   20.8 0.433 3.2 10  ⋯ 0.917 2   1  1  
    DRB RPG APG SPG BPG PPG
201 2.2 3.2 7   1.2 0.2 30

The player with the most Points Per Game in April is Damian Lillard 

## Disscussion

Overall this study was able to find the NBA player who scored the most points per game depending on the month in the 2017-2018 season. In October, the first month of the season the NBA player to score the most points per game was Giannis Antetokounmpo. In November and December, it was James Harden. January the player with the highest points per game was Stephen Curry, February it was Anthony Davis, March it was Lebron James, and finally, in April the player who scored the most points per game was Damian Lillard. Finding these results allows the study to conclusively show which players contribute the most to their team depending on the month. The study even further showed other findings such as a trend in which James Harden can maintain his high PPG for two months within the middle of the NBA season. Another compelling component from this study was it was able to find the highest PPG within the regular season was 35 obtained in February by Anthony Davis and that the highest PPG values ranged from 35 to 29.5.

This data is scientifically sound in which it is accurately able to find the maximum value in the PPG column within the data frame and take that information to find the name corresponding to the value. Some errors that may occur, however, could be due to inaccuracy of the website that is being used to gather the information. If the website is inaccurate, then the answers the coding provides are no longer valid. Another error could occur if the proper selling and names are misplaced or misspelled so much as the misuse of a capital letter where there should be a lowercase letter within any line of code in the study will cause the code not to work properly and thus not find the particular data that is being searched for.

This study can be used to find the maximum value for other columns within this specific table, similar statistics for any other sport, or even any additional information from a website that an individual may wish to research. Since the method is straightforward involved extracting HTML data and finding a maximum value, it is a widely applicable approach that can be used as an individual sees fit. A similar method was found in a blog post by a user named Alice Sweeting in which they wished to put data straight into R rather than what they were doing previous which was using a program called “Smartabase” to take data and produce a visualization and then import that data into R using a .csv file (Sweeting, 2018,). They also used HTML coding to find information within NBA statistics and showed the methods to take information from a website and use it to create a data set much like what has been produced in this study. David Smith conducted another approach similar to this one in with they found movement and play by play data and placed it within R to animate plays (Smith, 2016). While this study did not use HTML reading to find what this study was looking to see it still is significant because it shows different takes on importing sports data, specifically NBA data, as well as what can be done with it and why it is important.

## References

Bhandari I, Colet E, Parker J, Pines Z, Pratap R, Ramanujam. 1997. Advanced Scout: Data Mining and Knowledge Discovery in NBA Data. Data Min Knowl Disc. 1(1): 121-125.

Elmore R, DeWitt P. 2018 Nov 1. Ballr; [accessed 2018 Nov 9]. https://github.com/rtelmore/ballr/blob/master/README.md

Smith D. 2016 Sep 2. Analyzing NBA Basketball Data with R [blog]. Revolutions. [accessed 2018 Nov 9]; https://blog.revolutionanalytics.com/2016/09/analyzing-nba-basketball-data-with-r.html

Sweeting A. 2018 Sep 5. Quickly Importing Athlete Management System (AMS) Data into R [blog]. Sport Statistics R Sweet! [accessed 2018 Nov 9]; https://sportstatisticsrsweet.wordpress.com

Wickham H. 2014 Nov 24. Rvest: Easy Web Scraping with R [blog]. Rstudio [accessed 2018 Nov 9]; https://blog.rstudio.com/2014/11/24/rvest-easy-web-scraping-with-r/

Wickham H. 2017 June 13. Dplyr 0.70. [blog]. Rstudio [accessed 2018 Nov 9]; https://blog.rstudio.com/2017/06/13/dplyr-0-7-0/